analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LockedFieldPrivateReadonly.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Linq;
+
+class Test
+{
+    private static readonly object staticReadonlyField = null;
+    private static object staticReadWriteField = null;
+
+    private readonly object readonlyField = null;
+    private readonly string readonlyStringField = "a string";
+    private object readWriteField = null;
+
+    private static object StaticReadonlyProperty => null;
+    private object ReadonlyProperty => null;
+
+    private static object StaticReadWriteProperty { get; set; }
+    private object ReadWriteProperty { get; set; }
+
+    void OnAStaticField()
+    {
+        lock (staticReadonlyField) { }
+        lock (staticReadWriteField) { }                  // Noncompliant {{'staticReadWriteField' is not 'private readonly', and should not be used for locking.}}
+        //    ^^^^^^^^^^^^^^^^^^^^
+        lock (Test.staticReadonlyField) { }
+        lock (Test.staticReadWriteField) { }             // Noncompliant {{'staticReadWriteField' is not 'private readonly', and should not be used for locking.}}
+        //    ^^^^^^^^^^^^^^^^^^^^^^^^^
+        lock (AnotherClass.staticReadonlyField) { }      // Noncompliant {{'staticReadonlyField' is not 'private readonly', and should not be used for locking.}}
+        lock (AnotherClass.staticReadWriteField) { }     // Noncompliant {{'staticReadWriteField' is not 'private readonly', and should not be used for locking.}}
+        //    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+    }
+
+    void OnAFieldOfSameInstance()
+    {
+        lock (readonlyField) { }
+        lock (readonlyStringField) { }
```

Comment:
I'd argue that this should be non-compliant because of String Interning.
Another method can lock on a different string object with the same value, and if they are interned they will point to the same memory location, so they can unintentionally block each other.

Reply:
Indeed, fixed.